src/plugins/obj/obj_engine_registry.cpp

```
@@ -0,0 +1,54 @@
+
+std::unique_ptr<nixlObjEngineImpl>
+objAccelEngineRegistry::create(const std::string &type,
+                               const nixlBackendInitParams *params) const {
+    auto it = entries_.find(type);
+    if (it != entries_.end())
+        return it->second.create(params);
+
+    if (type.empty())
+        throw std::runtime_error("Accelerated engine support not available (not compiled)");
+
+    throw std::runtime_error(
```

Cppcheck (2.20.0)

[style] The function 'waitUntilIdle' is never used.

(unusedFunction)